File: ios/Kurir/modules/tasks/TasksViewModel.swift
//
//  FindViewModel.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Crazybean
import Mobilex

class TasksViewModel: ViewModel {
    private let userData: UserData?
    private let usersRepository: UsersRepository
    private let tasksRepository: TasksRepository
    
    init(userData: UserData?, usersRepository: UsersRepository, tasksRepository: TasksRepository) {
        self.userData = userData
        self.usersRepository = usersRepository
        self.tasksRepository = tasksRepository
    }
    
    func loadTasks() -> LiveData<[Task]?> {
        return LiveData<[Task]?>() { [weak self] liveData in
            self?.tasksRepository.getTasks(picker: nil, completion: { tasks in
                liveData.value = tasks
            })
        }
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/tasks/TasksViewModel.kt
package au.com.crazybean.mobilex.kurir.modules.tasks

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import au.com.crazybean.foundation.mvvm.ViewModel
import au.com.crazybean.mobilex.foundation.userdata.UserData
import au.com.crazybean.mobilex.kurir.data.model.Task
import au.com.crazybean.mobilex.kurir.repository.tasks.TasksRepository
import au.com.crazybean.mobilex.kurir.repository.users.UsersRepository

class TasksViewModel(private val userData: UserData?,
                     private val usersRepository: UsersRepository?,
                     private val tasksRepository: TasksRepository?) : ViewModel() {

    val tasks: LiveData<List<Task>?>
        get() = MutableLiveData<List<Task>?>().also { liveData ->
            tasksRepository?.getTasks { tasks ->
                liveData.value = tasks
            }
        }
}